model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -72,6 +73,31 @@ import org.modelix.model.server.store.LocalModelClient
 import org.modelix.modelql.server.ModelQLServer
 import org.slf4j.LoggerFactory
 
+private class BranchNotFoundException(branch: String, repositoryId: String) : HttpException(
+    HttpStatusCode.NotFound,
+    title = "Branch not found",
+    details = "Branch '$branch' does not exist in repository '$repositoryId'",
+    type = "/problems/type-not-found",
+) {
+    constructor(branch: BranchReference) : this(branch.branchName, branch.repositoryId.id)
+}
+
+private class VersionNotFoundException(versionHash: String) :
+    HttpException(
+        HttpStatusCode.NotFound,
+        title = "Version not found",
+        details = "Version '$versionHash' doesn't exist",
+        type = "/problems/version-not-found",
+    )
+
+private class BadRequestException(details: String, typeSuffix: String, cause: Throwable? = null) : HttpException(
+    HttpStatusCode.BadRequest,
+    title = "Bad request",
+    details = details,
+    type = "/problems/$typeSuffix",
+    cause = cause,
+)
+
```

Comment:
Can we move these Exceptions into a separate file? They are not specific to `ModelReplicationServer` and could be reused in other places like the `ContentExplorer`.

Reply:
These exceptions are closely related to the REST API. So they would need to stay in the `handlers` package.